data-model/src/main/java/io/micronaut/data/model/PersistentEntity.java

```diff
@@ -88,6 +87,20 @@ default boolean hasIdentity() {
      */
     @Nullable PersistentProperty getIdentity();
 
+    /**
+     * Returns all identity properties.
+     *
+     * @return The identity properties
+     */
+    @NonNull
+    default List<PersistentProperty> getIdentityProperties() {
+        if (getIdentity() != null) {
```

Comment:
Should look at `getCompositeIdentity()` ?

Reply:
I will try to add it an see if it's going to work

Reply:
I guess check if getCompositeIdentity is not empty then return it, else list with getIdentity or else empty list
To have same logic as implementing classes I guess.